### ISSUE TITLE
Kafka consumer, lag on topic metric unit is wrong?

### ISSUE THREAD

**Issue by saunaCat** [spring-cloud/spring-cloud-stream-binder-kafka#446]
The readme says the metric spring.cloud.stream.binder.kafka.offset "indicates how many messages have not been yet consumed from a given binder’s topic by a given consumer group.", but it's given in milliseconds.  Is it really supposed to be a TimeGauge measurement?

https://github.com/spring-cloud/spring-cloud-stream-binder-kafka/blob/e869516e3d0577db98d9c81e34e5e94e6200b7c7/spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaBinderMetrics.java#L114

**Comment by sobychacko** [comment]
@saunaCat This metric is published into a micrometer `MeterRegistry`. In your application, you can inject `MeterRegistry` bean and call `find` and `get` methods to search for the metrics. Not sure, if this is what you were looking. In any case, please use [gitter](https://gitter.im/spring-cloud/spring-cloud-stream) or [SO](https://stackoverflow.com/questions/tagged/spring-cloud-stream) for this kind of questions. We prefer GitHub issues for feature requests or bug reports.

**Comment by porshkevich** [comment]
> @saunaCat This metric is published into a micrometer `MeterRegistry`. In your application, you can inject `MeterRegistry` bean and call `find` and `get` methods to search for the metrics. Not sure, if this is what you were looking. In any case, please use [gitter](https://gitter.im/spring-cloud/spring-cloud-stream) or [SO](https://stackoverflow.com/questions/tagged/spring-cloud-stream) for this kind of questions. We prefer GitHub issues for feature requests or bug reports.

I'm not sure that you understood the question correctly. Since I have the same question, let me rephrase.

Why used TimeGauge in KafkaBinderMetrics??? 

The Metric spring.cloud.stream.binder.kafka.offset indicates: "how many messages have not been yet consumed from a given binder’s topic by a given consumer group". Kafka offset it`s just "Counter" from 0 to MAX (in micrometer Gauge) but not a Time.

I found a pull request, creating this mess. https://github.com/spring-cloud/spring-cloud-stream-binder-kafka/pull/325

I do not understand how it was missed, BUT kafka offset measure not by time.

**Comment by olegz** [comment]
The real issue here is the naming of `calculateConsumerLagOnTopic(..)` operation, which is confusing and created an appearance of some time-based measure, so I'll fix it during the merge.
Also, @porshkevich in the future please tone down the rhetoric. Calling something a _"mess"_ is meaningless, counterproductive and thus won't get you anywhere. Other thank that thank you for discovering and fixing a  _"bug"_.
